docs/source/modules/nn.rst

```diff
@@ -42,6 +42,96 @@ Aggregation Operators
+   mean_aggr = aggr.MeanAggregation()
+   max_aggr = aggr.MaxAggregation()
+
+   # Advanced aggregations:
+   median_aggr = aggr.MedianAggregation()
+
+   # Learnable aggregations:
+   softmax_aggr = aggr.SoftmaxAggregation(learn=True)
+   powermean_aggr = aggr.PowerMeanAggregation(learn=True)
+
+   # Exotic aggregations:
+   lstm_aggr = aggr.LSTMAggregation(in_channels=..., out_channels=...)
+   sort_aggr = aggr.SortAggregation(k=4)
+
+   x = torch.randn(1000, 64)  # 1000 nodes with 64 features each.
+   index = torch.randint(0, 100, (1000, ))  # Assign nodes to 100 graphs.
+
+   output = mean_aggr(x, index)  #  Output shape: [100, 64]
+
+Notably, all aggregations share the same set of forward arguments, as described in detail in :meth:`torch_geometric.nn.aggr.Aggregation.forward`.
+
+Each of the provided aggregations can be used within :class:`~torch_geometric.nn.conv.MessagePassing` as well as for hierachical/global pooling to obtain graph-level representations:
+
+.. code-block:: python
+
+   import torch
+   from torch_geometric.nn import MessagePassing
+
+   class MyConv(MessagePassing):
+       def __init__(self, ...):
+           # Use a learnable softmax neighborhood aggregation:
+           super().__init__(aggr=aggr.SoftmaxAggregation(learn=True))
+
+      def forward(self, x, edge_index):
+          ....
+
+
+   class MyGNN(torch.nn.Module)
+       def __init__(self, ...):
+           super().__init__()
+
+           self.conv = MyConv(...)
+           # Use a global sort aggregation:
+           self.global_pool = aggr.SortAggregation(k=4)
+           self.classifier = torch.nn.Linear(...)
+
+        def foward(self, x, edge_index, batch):
+            x = self.conv(x, edge_index).relu()
+            x = self.global_pool(x, batch)
+            x = self.classifier(x)
+            return x
+
+In addition, the aggregation package of PyG introduces two new concepts:
+First, aggregations can be **resolved from pure strings** via a lookup table, following the design principles of the `class-resolver <https://github.com/cthoyt/class-resolver>`__ library, *e.g.*, by simply passing in :obj:`"median"` to the :class:`~torch_geometric.nn.conv.MessagePassing` module.
+This will automatically resolve to the :obj:`~torch_geometric.nn.aggr.MedianAggregation` class:
+
+.. code-block:: python
+
+   class MyConv(MessagePassing):
+       def __init__(self, ...):
+           super().__init__(aggr="median")
+
+Secondly, **multiple aggregations** can be combined and stacked via the :class:`~torch_geometric.nn.aggr.MultiAggregation` module in order to enhance the representational power of GNNs (`Corso et al. (2020) <https://arxiv.org/abs/2004.05718>`__, `Tailor et al. (2021) <https://arxiv.org/abs/2104.01481>`__):
+
+.. code-block:: python
+
+   class MyConv(MessagePassing):
+       def __init__(self, ...):
+           super().__init__(aggr=aggr.MultiAggregation(
+               ['mean', 'std', aggr.SoftmaxAggregation(learn=True)]))
```

Comment:
Perhaps add a comment above indicating that this is supported due to automatic resolution? This is indicated in the sentence below, but may be worth adding a brief comment here since it is pretty cool and somewhat unexpected